data/mods/gen8bdsp/random-teams.ts

```diff
@@ -1,7 +1,162 @@
 
-export class RandomBDSPTeams extends RandomTeams {}
+export class RandomBDSPTeams extends RandomTeams {
+	constructor(format: Format | string, prng: PRNG | PRNGSeed | null) {
+		super(format, prng);
+	}
+
+	getHighPriorityItem(
+		ability: string,
+		types: Set<string>,
+		moves: Set<string>,
+		counter: MoveCounter,
+		teamDetails: RandomTeamsTypes.TeamDetails,
+		species: Species,
+		isLead: boolean,
+		isDoubles: boolean
+	) {
+		// not undefined — we want "no item" not "go find a different item"
+		if (moves.has('acrobatics')) return '';
+		if (moves.has('solarbeam') && !(moves.has('sunnyday') || teamDetails.sun)) return 'Power Herb';
+		if (moves.has('shellsmash')) return 'White Herb';
+		// Species-specific item assigning
+		if (species.name === 'Farfetch\u2019d') return 'Leek';
+		if (species.name === 'Latios' || species.name === 'Latias') return 'Soul Dew';
+		if (species.name === 'Lopunny') return 'Toxic Orb';
+		if (species.baseSpecies === 'Marowak') return 'Thick Club';
+		if (species.baseSpecies === 'Pikachu') return 'Light Ball';
+		if (species.name === 'Shedinja') return 'Focus Sash';
+		if (species.name === 'Shuckle' && moves.has('stickyweb')) return 'Mental Herb';
+		if (['Sniper', 'Super Luck'].includes(ability) || moves.has('focusenergy')) return 'Scope Lens';
+		if (species.name === 'Wobbuffet' && moves.has('destinybond')) return 'Custap Berry';
+		if (species.name === 'Scyther' && counter.damagingMoves.size > 3) return 'Choice Band';
+		if ((moves.has('bellydrum') || moves.has('tailglow')) && moves.has('substitute')) return 'Salac Berry';
+
+		// Ability based logic and miscellaneous logic
+		if (species.name === 'Wobbuffet' || ability === 'Ripen') return 'Sitrus Berry';
+		if (ability === 'Gluttony') return this.sample(['Aguav', 'Figy', 'Iapapa', 'Mago', 'Wiki']) + ' Berry';
+		if (ability === 'Imposter') return 'Choice Scarf';
+		if (ability === 'Guts' && counter.get('Physical') > 2) {
+			return types.has('Fire') ? 'Toxic Orb' : 'Flame Orb';
+		}
+		if (ability === 'Magic Guard' && counter.damagingMoves.size > 1) {
+			return moves.has('counter') ? 'Focus Sash' : 'Life Orb';
+		}
+		if (ability === 'Sheer Force' && counter.get('sheerforce')) return 'Life Orb';
+		if (ability === 'Unburden') return (moves.has('closecombat') || moves.has('curse')) ? 'White Herb' : 'Sitrus Berry';
+
+		if (moves.has('trick') || moves.has('switcheroo')) {
+			if (species.baseStats.spe >= 60 && species.baseStats.spe <= 108 && !counter.get('priority')) {
+				return 'Choice Scarf';
+			} else {
+				return (counter.get('Physical') > counter.get('Special')) ? 'Choice Band' : 'Choice Specs';
+			}
+		}
+		if (moves.has('auroraveil') || moves.has('lightscreen') && moves.has('reflect')) return 'Light Clay';
+		if (moves.has('rest') && !moves.has('sleeptalk') && ability !== 'Shed Skin') return 'Chesto Berry';
+		if (moves.has('bellydrum')) return 'Sitrus Berry';
+	}
+
+	getMediumPriorityItem(
```

Comment:
what's changed in these item generation functions from Sw/Sh? Is it just removing the items that don't exist in BDSP?

Reply:
removing references of moves/items/abilities mostly